frontend/src/components/ConstraintForm/SubjectAccordionItem.vue

```diff
@@ -59,14 +74,16 @@ export default {
       <span>Teacher</span>
       <select v-model="subject.teacher.name" class="rsprd-select">
         <option value="">-</option>
-        <option v-for="teacher in teachers" :key="teacher">{{ teacher }}</option>
+        <option v-for="teacher in teachers" :key="teacher">
+          {{ formatTeacherOptionContent(teacher) }}
```

Comment:
I realize now my linking to the computed documentation made it seem this should now be a computed. This is basically the same situation as before. This computed does not depend directly on any `ref`s. They receive some through a prop but I do not believe that allows Vue to cache it.

In general, computed should return some value, rarely it should be a function. It should be looked at (in most cases) as equivalent to the values in `data`. But they are not editable and are reactive to changes to the `ref`s they use.

My original idea was this:
```js
const formatTeacherOptionContent = teacher => {
  teacher.displayName = `${teacher.teacherCode} - ${teacher.firstName} ${teacher.lastName}`;
  return teacher;
};

const loadTeachers = () => {
  teacherService
    .getAllTeachers()
    .then(response => response.map(formatTeacherOptionContent))
    .then(data => (teachers.value = data));
};
```
We know that the value of `teachers` only changes when they are fetched from the server. There is no need for reactivness. So the data can be parsed right after fetching.

But, since you used a computed, a realized it can be done in one additional way:
```js
const parsedTeachers = computed(() => {
  return teachers.value.map(teacher  => `${teacher.teacherCode} - ${teacher.firstName} ${teacher.lastName}`);
});
// or
const parsedTeachers = computed(() => {
  return teachers.value.map(teacher  => {
    const displayName = `${teacher.teacherCode} - ${teacher.firstName} ${teacher.lastName}`);
    return { ...teacher, displayName };
  });
});
```
Now this computed can be used in the `v-for` and it will react to any changes made to the teachers `ref`. This way is good when teachers `ref` can change in multiple ways or you do not want to touch the returned data for some reason.

The choice is yours which way to take it. Sorry for the confusion.